core/src/main/scala/chisel3/StrongEnum.scala

```diff
@@ -233,6 +233,9 @@ abstract class EnumFactory {
 
   private[chisel3] val enumTypeName = getClass.getName.init
 
+  // Do all bitvectors of this Enum's width represent legal states?
+  private[chisel3] def isTotal: Boolean = enumRecords.size == (1L << this.getWidth)
```

Comment:
I think `ArrayBuffer.size` returns an `Int`. So, there may be some weird behavior here for widths > 32 bits. I don't anticipate this actually coming up, though.

Reply:
Fair point on `.size`, I'm just in the habit of always using `Long` when left-shifting but I think it's irrelevant here. We could add an assertion that `this.getWidth` is < 31 and call it a day.

Reply:
Done by guarding against Integer overflow